tasks/pipeline.py

```diff
@@ -991,3 +996,72 @@ def test_merge_queue(ctx):
     ctx.run(f"git push origin :{test_main}", hide=True)
     if not success:
         raise Exit(message="Merge queue test failed", code=1)
+
+
+@task
+def compare_to_itself(ctx):
+    """
+    Create a new branch with 'compare_to_itself' in gitlab-ci.yml and trigger a pipeline
+    """
+    if not gitlab_configuration_is_modified(ctx):
+        print("No modification in the gitlab configuration, ignoring this test.")
+        return
+    agent = get_gitlab_repo()
+    gh = GithubAPI()
+    current_branch = os.getenv("CI_COMMIT_REF_NAME")
+    new_branch = f"compare/{current_branch}/{int(datetime.now().timestamp())}"
+    ctx.run(f"git checkout -b {new_branch}", hide=True)
+    ctx.run(
+        f"git remote set-url origin https://x-access-token:{gh._auth.token}@github.com/DataDog/datadog-agent.git",
+        hide=True,
+    )
+    ctx.run("git config --global user.name 'github-app[bot]'")
+    ctx.run("git config --global user.email 'github-app[bot]@users.noreply.github.com'")
+    # The branch must exist in gitlab to be able to "compare_to"
+    ctx.run(f"git push origin {new_branch}", hide=True)
+    time.sleep(15)  #  Wait for the branch to be created
```

Comment:
Is it necessary ? I think that the command will block until it is created (or maybe wait 1s to be sure)

Reply:
I've just added it because on some cases the first push event was not generated. But adding it put me in an infinite loop, I must ensure I won't generate a monster...

Reply:
I remove it and change slightlty the test below to make it more robust